packages/kit/src/core/sync/write_root.js

```diff
@@ -40,28 +40,57 @@ export function write_root(manifest_data, output) {
 		dedent`
 			<!-- This file is generated by @sveltejs/kit — do not edit it! -->
 			<script>
-				import { setContext, afterUpdate, onMount, tick } from 'svelte';
+				import { setContext, ${isSvelte5Plus() ? '' : 'afterUpdate, '}onMount, tick } from 'svelte';
 				import { browser } from '$app/environment';
 
 				// stores
-				export let stores;
-				export let page;
-
-				export let constructors;
-				export let components = [];
-				export let form;
-				${levels.map((l) => `export let data_${l} = null;`).join('\n')}
+				${
+					isSvelte5Plus()
+						? dedent`
+							let { stores, page, constructors, components = [], form, ${levels
+								.map((l) => `data_${l} = null`)
+								.join(', ')} } = $props();
+						`
+						: dedent`
+							export let stores;
+							export let page;
+
+							export let constructors;
+							export let components = [];
+							export let form;
+							${levels.map((l) => `export let data_${l} = null;`).join('\n')}
+						`
+				}
 
 				if (!browser) {
 					setContext('__svelte__', stores);
 				}
 
-				$: stores.page.set(page);
-				afterUpdate(stores.page.notify);
+				${
+					isSvelte5Plus()
+						? dedent`
+							if (browser) {
+								$effect.pre(() => stores.page.set(page));
+							} else {
+								stores.page.set(page);
```

Comment:
would this not work in the browser case as well?

Reply:
No, effects don't run in SSR. This is the rare case where the code inside a `$: {...}` _does_ need to run in SSR, but it's because we're having to do things in a bit of a roundabout way to begin with. This `if` is the correct way to deal with it